bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
+
+    def efferent_edges(self, node_id, properties=None):
+        """Get efferent edges for given ``node_id``.
+
+        Args:
+            node_id: source node ID
+            properties: None / edge property name / list of edge property names
+
+        Returns:
+            List of edge IDs, if ``properties`` is None;
+            Pandas Series indexed by edge IDs if ``properties`` is string;
+            Pandas DataFrame indexed by edge IDs if ``properties`` is list.
+        """
+        return self.pathway_edges(source=node_id, target=None, properties=properties)
+
+    def pair_edges(self, source_node_id, target_node_id, properties=None):
+        """Get edges corresponding to ``source_node_id`` -> ``target_node_id`` connection.
+
+        Args:
+            source_node_id: source node ID
+            target_node_id: target node ID
+            properties: None / edge property name / list of edge property names
+
+        Returns:
+            List of edge IDs, if ``properties`` is None;
+            Pandas Series indexed by edge IDs if ``properties`` is string;
+            Pandas DataFrame indexed by edge IDs if ``properties`` is list.
+        """
+        return self.pathway_edges(
+            source=source_node_id, target=target_node_id, properties=properties
+        )
+
+    @staticmethod
+    def _add_circuit_ids(its, source, target):
+        """Generator comprehension adding the CircuitIds to the iterator.
+
+        Notes:
+            Using closures or lambda functions would result in override functions and so the
+            source and target would be the same for all the populations.
+        """
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1]), it[2]) for it in its)
+
+    @staticmethod
+    def _add_edge_ids(its, source, target, pop_name):
+        """Generator comprehension adding the CircuitIds to the iterator."""
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1]),
+                 CircuitEdgeIds.from_dict({pop_name: it[2]})) for it in its)
+
+    @staticmethod
+    def _omit_edge_count(its, source, target):
+        """Generator comprehension adding the CircuitIds to the iterator."""
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1])) for it in its)
+
+    def iter_connections(
+            self, source=None, target=None, return_edge_ids=False,
+            return_edge_count=False):
+        """Iterate through ``source`` -> ``target`` connections.
+
+        Args:
+            source (CircuitNodeIds/int/sequence/str/mapping/None): source node group
+            target (CircuitNodeIds/int/sequence/str/mapping/None): target node group
+            return_edge_count: if True, edge count is added to yield result
+            return_edge_ids: if True, edge ID list is added to yield result
+
+        ``return_edge_count`` and ``return_edge_ids`` are mutually exclusive.
```

Comment:
You know that I will ask this. So I will. Why not 3 different functions ?

Also:
You don't need to start this function by checking that both are not True

Reply:
Done for the both are True.

About the 3 functions this is again to keep the historical behavior. I agree with you it should not be like that. One day I will clean this up, I don t like it to much also and find it perturbing.